core/domain/recommendations_jobs_one_off.py

```diff
@@ -124,3 +126,44 @@ def map(model):
     @staticmethod
     def reduce(key, values):
         yield (key, len(values))
+
+
+class CleanupExplorationRecommendationsOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Cleans up ExplorationRecommendationsModel by:
+    1. Removing exploration ids from recommendation list of model if exploration
+    model is deleted.
+    2. Deleting the recommendations model if the exploration for which it was
+    created is deleted.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [recommendations_models.ExplorationRecommendationsModel]
+
+    @staticmethod
+    def map(item):
+        exp_model = exp_models.ExplorationModel.get_by_id(item.id)
+        if exp_model is None or exp_model.deleted:
+            item.delete()
+            yield ('Removed recommendation model', 1)
```

Comment:
Yield the ID and don't aggregate, this is a failure / unexpected case.

Use counts only for successes / happy-path.

Reply:
Done.